requre/objects.py

```diff
@@ -62,14 +62,30 @@ def __init__(
     ) -> None:
         self.store_keys = store_keys
         if cassette:
-            self.cassette = cassette
+            self.set_cassette(cassette)
         self.store_keys = store_keys
         self.storage_object_kwargs = storage_object_kwargs or {}
 
+    @classmethod
+    def get_cassette(cls):
+        """
+        Internal method to return proper cassette, it is important after rewriting of singletons
+        Cassette were not passed to object executor
+        :return: Cassette instance
+        """
+        if cls._cassette:
+            return cls._cassette
+        else:
+            return PersistentObjectStorage().cassette
```

Comment:
This looks like another singleton. Is this really needed? Cannot we get rid of the class methods and work with single instances? We can avoid lot of problems if we avoid sharing state between instances.

Reply:
it is there because of backward compatibility, in case I'll remove it, I have to remove all occurences inside other projects, ogr, packit and packit-service and completely rewrite it to new structure. Theoretically I can remove it after all other projects get rid of them.

Reply:
OK, can we raise a deprecation warning, please?

Reply:
Looks like you are fixing a lot of independent problems here.

Reply:
seems independent, but were raised when I've changed the code and found that there is strange behaviour. Yes I could raise deprecation waring when using singleton, but probably I'll do that after all fixes in all our projests, to not have there lots of warnings when tests are executed.